src/FactoryTrait.php

```diff
@@ -54,10 +67,26 @@ public function factory($object, $defaults = [])
      */
     public function normalizeClassName($name, $prefix = null)
     {
+        if (is_array($name) && isset($name[0])) {
+            $name[0] = $this->normalizeClassName($name[0]);
+
+            return $name;
+        }
+
         if (!is_string($name)) {
             return $name;
         }
 
+        if (isset($this->_appScopeTrait) && $this->app) {
```

Comment:
this can be moved slightly up and can use already existing `if (!is_string($name)) return $name;` part above.